canonicalwebteam/discourse/parsers/docs.py

```diff
@@ -474,9 +474,14 @@ def _parse_version_table(self, main_index_soup):
         tables = navigation_soup.findAll("table")
 
         for table in tables:
-            if table.select("tr:has(> th:-soup-contains('Version'))"):
-                version_table = table.select("tr:has(td)")
-
+            first_row = table.tr
+            if not first_row:
+                continue
+            headers = first_row("th")
+            if not headers:
+                continue
+            if headers[-1].string == "Version":
+                version_table = table("tr")[1:]
```

Comment:
This looks a bit loose compared to `table.select("tr:has(td)")`

Reply:
Right, so the previous code was something like


IF the table has a <tr> which has a <th> which contains "Version" THEN set version_table to the rows which contain <td>

then below (in the `for row in version_table` loop) it gets `path` and `version_cell` both as `row.select_one("td:...")` which ensures that it's only ever looking at <td>s

This is why it's safe (and about 10x faster) to get all but the first of the <tr>s as this branch does